src/dataset.js

```diff
@@ -24,6 +25,73 @@ const getLocaleFilename = index => `${leftpad(index, LOCAL_FILENAME_DIGITS, 0)}.
 const { datasets } = apifyClient;
 const datasetsCache = new LruCache({ maxLength: MAX_OPENED_STORES }); // Open Datasets are stored here.
 
+/**
+ * Accepts a JSON serializable object as an input, validates its serializability,
+ * and validates its serialized size against limitBytes. Optionally accepts its index
+ * in an array to provide better error messages. Returns serialized object.
+ *
+ * @param {Object} item
+ * @param {Number} limitBytes
+ * @param {Number} [index]
+ * @returns {string}
+ * @ignore
+ */
+export const checkAndSerialize = (item, limitBytes, index) => {
+    const s = typeof index === 'number' ? ` at index ${index} ` : ' ';
+    let payload;
+    try {
+        payload = JSON.stringify(item);
+    } catch (err) {
+        throw new Error(`Data item${s}is not serializable to JSON.`);
```

Comment:
We need to show an error otherwise user has no idea what has happened, what about:

`Data item${s}cannot be stringified to JSON: ${e.message}`